dsme/utility.c

```diff
@@ -64,6 +66,24 @@ dsme_user_is_privileged(uid_t uid, gid_t gid)
 {
     bool is_privileged = false;
 
+#ifdef DSME_USEWHEEL
+    struct passwd* pw = getpwuid(uid);
+    if( pw ) {
+        int ngroups = 0;
+        getgrouplist(pw->pw_name, pw->pw_gid, NULL, &ngroups);
+        gid_t groups[ngroups];
+        getgrouplist(pw->pw_name, pw->pw_gid, groups, &ngroups);
+
+        for (int i = 0; i < ngroups; i++) {
+            struct group* gr = getgrgid(groups[i]);
+            if( gr != NULL ) {
+                if(strcmp(gr->gr_name,"wheel") != 0 ) {
+                   is_privileged = true;
```

Comment:
All groups except "wheel" are privileged? Should be `if( strcmp() == 0 )`?

Also, once a match is found, one might as well break out of the loop.